packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -1058,3 +1065,273 @@ class CallbackShortcuts extends StatelessWidget {
+///
+/// The registry may be listened to (with [addListener]/[removeListener]) for
+/// change notifications when the registered shortcuts change.
+class ShortcutRegistry extends ShortcutManager {
+  /// Gets the combined shortcut bindings from all contexts that are registered
+  /// with this [ShortcutRegistry], in addition to the bindings passed to
+  /// [ShortcutRegistry.ShortcutRegistry].
+  ///
+  /// Listeners will be notified when the value returned by this getter changes.
+  ///
+  /// Returns a copy: modifying the returned map will have no effect.
+  @override
+  Map<ShortcutActivator, Intent> get shortcuts {
+    assert(debugAssertNotDisposed());
+    return <ShortcutActivator, Intent>{
+      ..._shortcuts,
+      for (final MapEntry<ShortcutRegistryToken, Map<ShortcutActivator, Intent>> entry in _tokenShortcuts.entries)
+        ...entry.value,
+    };
+  }
+  final Map<ShortcutRegistryToken, Map<ShortcutActivator, Intent>> _tokenShortcuts =
+    <ShortcutRegistryToken, Map<ShortcutActivator, Intent>>{};
+
+  /// Adds all the given shortcut bindings to this [ShortcutRegistry], and
+  /// returns a token for managing those bindings.
+  ///
+  /// The token should have [ShortcutRegistryToken.dispose] called on it when
+  /// these shortcuts are no longer needed. This will remove them from the
+  /// registry, and invalidate the token.
+  ///
+  /// This method will assert in debug mode if another token exists (i.e. hasn't been
+  /// disposed of) has already added a given shortcut.
+  ///
+  /// If two equivalent, but different, [ShortcutActivator]s are added, all of
+  /// them will be executed when triggered. For example, if both
+  /// `SingleActivator(LogicalKeyboardKey.keyA)` and `CharacterActivator('a')`
+  /// are added, then both will be executed when an "a" key is pressed.
+  ShortcutRegistryToken addAll(Map<ShortcutActivator, Intent> value) {
+    assert(debugAssertNotDisposed());
+    final ShortcutRegistryToken token = ShortcutRegistryToken._(this);
+    _tokenShortcuts[token] = value;
+    _debugCheckForDuplicates();
+    _indexedShortcutsCache = null;
+    notifyListeners();
+    return token;
+  }
+
+  /// Replaces the given shortcut bindings in this [ShortcutRegistry].
+  ///
+  /// This method will assert in debug mode if another [ShortcutRegistryToken]
+  /// exists (i.e. hasn't been disposed of) that has already added a given
+  /// shortcut.
+  ///
+  /// It will also assert if the given token wasn't created by this registry, or
+  /// has already been disposed.
+  ///
+  /// If two equivalent, but different, [ShortcutActivator]s are added, all of
+  /// them will be executed when triggered. For example, if both
+  /// `SingleActivator(LogicalKeyboardKey.keyA)` and `CharacterActivator('a')`
+  /// are added, then both will be executed when an "a" key is pressed.
+  void replaceAll(ShortcutRegistryToken token, Map<ShortcutActivator, Intent> value) {
```

Comment:
In theory, this could be a method on the token itself, similarly to how dispose is done...

Reply:
Yeah, OK.  Done.